# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/UtilityProcessChild.h
# Commit: d56484304ec3
# Full Hash: d56484304ec3abe3891ede9e8d141c95543d3e37
# Author: Ray Kraesig <rkraesig@mozilla.com>
# Date: 2023-09-16 09:14:45
# Regressor Bug: 1837079
# File Overlap Count: 2
# Description:
#   Bug 1837079 - [5/10] Create WinFileDialog IPDL protocol and implementation  r=gstoll,handyman,ipc-reviewers,nika
#   
#   Create and implement a new top-level IPC protocol, `PWinFileDialog`,
#   using the primitives from bug 1833450 and the new sandboxing type from
#   the previous commit.
# ==============================================================================

diff -r 36d3afd881b4 -r d56484304ec3 ipc/glue/UtilityProcessChild.h
--- a/ipc/glue/UtilityProcessChild.h	Fri Sep 15 18:10:00 2023 +0000
+++ b/ipc/glue/UtilityProcessChild.h	Fri Sep 15 18:10:01 2023 +0000
@@ -72,6 +72,9 @@
   mozilla::ipc::IPCResult RecvStartWindowsUtilsService(
       Endpoint<PWindowsUtilsChild>&& aEndpoint);
 
+  mozilla::ipc::IPCResult RecvStartWinFileDialogService(
+      Endpoint<PWinFileDialogChild>&& aEndpoint);
+
   mozilla::ipc::IPCResult RecvGetUntrustedModulesData(
       GetUntrustedModulesDataResolver&& aResolver);
   mozilla::ipc::IPCResult RecvUnblockUntrustedModulesThread();
@@ -97,6 +100,7 @@
   RefPtr<dom::JSOracleChild> mJSOracleInstance{};
 #ifdef XP_WIN
   RefPtr<PWindowsUtilsChild> mWindowsUtilsInstance;
+  RefPtr<PWinFileDialogChild> mFileDialogInstance;
 #endif
 
   AsyncBlockers mShutdownBlockers;